### DISCUSSION TITLE
Mocking table instance for testing

### DISCUSSION THREAD

**Discussion by andreagari** [KevinVandy/material-react-table#1198] [Q&A]
Hi!
I have a component who receives the table instance on props. Is there any way to mock that table, to be able to test that component?

This is a short example of the component definition:

type TComponentProps<T extends MRT_RowData> = {
  table: MRT_TableInstance<T>
  prop2: any[
  prop3:any
}

const Component = <T extends MRT_RowData>({
  table,
  prop2,
  prop3,
}: TComponentProps<T>) => {

.....

return <></> 
}

Thanks for the support!

**Comment by KevinVandy** [comment]
Probably don't mock a table instance, just actually create one.

**Reply by andreagari** [reply]
Hi @KevinVandy, thanks for answer me!!
I tried to create a new one into the test properly, but i cant use the  createTable hook inside it, thats why I was asking for a mock example...or any other idea?

Thanks!
